### ISSUE TITLE
[ic-popover-menu] - VoiceOver keyboard navigation bug

### ISSUE THREAD

**Issue by GCHQ-Developer-741** [mi6/ic-ui-kit#1900]
## Summary of the bug 
When trying to navigate through ic-popover-menu using a keyboard in combination with VoiceOver, it is not possible to navigate and click on the back button on any subMenu. The keyboard will skip over it and instead select the whole popover group. Can still navigate and click on other menu items.

## 🪜 How to reproduce 
1. Go to the ic-popover-menu storybook
2. Click on the first example
3. Open up VoiceOver
4. Navigate to the second subMenu
5. See the error

## 📸 Screenshots or code 

https://github.com/mi6/ic-ui-kit/assets/117368893/11dad939-a1a3-48b3-8d43-65f857fb07e0

## 🖥 📱 Device 
- Type: Desktop
- Device: MacBook
- Browser version: Firefox 126, Chrome 125

## 🧐 Expected behaviour 
The back button should be focusable and clickable by keyboard, as it is when not using VoiceOver.

## Additional info 
There are also a few quirks and differences when navigating the popover using a keyboard & VoiceOver vs not using VoiceOver. For example, I could only navigate through the menu items using the left/right arrow keys, and clicking involved ctrl + opt + space. These may just be VoiceOver intended behaviour, however the keyboard experience is much smoother when not using VoiceOver.

**Comment by GCHQ-Developer-299** [comment]
The back button **is** accessible to screen readers, it's just at the bottom of the menu in the tab order (same for keyboard and for voiceover)

this feels deliberate, i'm assuming we wanted the back button to be the last thing someone tabs through - but it does cause a misalignment between visuals and nav order. This is potentially bad practice and maybe a WCAG failure, i'll have a look...
